adr/0026-schema.md

```diff
@@ -0,0 +1,578 @@
+
+
+Below is an example v1 zarf.yaml with, somewhat, reasonable & nonempty values for every key
+```yaml
+kind: ZarfPackageConfig
+apiVersion: v1
+metadata:
+  name: everything-zarf-package
+  description: A zarf package with a non empty value for every
+  version: v1.0.0
+  url: https://my-package-website.com
+  image: https://my-image-url-to-use-in-deprecated-zarf-ui # TODO This field should be deprecated
+  uncompressed: true
+  architecture: amd64
+  yolo: false
+  authors: cool-kidz
+  documentation: https://my-package-documentation.com
+  source: https://my-git-server/my-package
+  vendor: my-vendor
+  aggregateChecksum: shasum # created by Zarf, probably should be moved to the build section
+build: # Everything here is created by Zarf not be users
+  terminal: my-computer
+  user: my-user
+  architecture: amd64
+  timestamp: 2021-09-01T00:00:00Z
+  version: v1.0.0
+  migrations:
+    - scripts-to-actions
+  registryOverrides:
+    gcr.io: my-gcr.com
+  differential: true
+  differentialPackageVersion: "v0.99.9"
+  differentialMissing:
+    - missing-component
+  flavor: cool-flavor
+  lastNonBreakingVersion: "v0.99.9"
+components:
+- name: a-component
+  description: Zarf description
+  default: false # Austin to check if we remove this
+  only:
+    localOS: darwin
+    cluster:
+      architecture: amd64
+      distros:
+      - ubuntu
+    flavor: a-flavor # this will only be used when there are multiple components
+  import:
+    name: other-component-name
+    path: ABCD # Only path or URL will be used, not both
+    url: oci://
+  manifests:
+  - name: manifest
+    namespace: manifest-ns
+    files:
+    - a-file.yaml
+    kustomizeAllowAnyDirectory: false
+    kustomizations:
+    - a-kustomization.yaml
+    noWait: false
+  charts:
+  - name: chart
+    namespace: chart-ns
+    version: v1.0.0
+    url: https://chart-url.com # Can only have one of url or localPath
+    localPath: folder
+    repoName: repo # We should change these names to make them less confusing https://github.com/defenseunicorns/zarf/issues/2245
+    gitPath: charts/podinfo
+    releaseName: chart-release
+    noWait: true
+    valuesFiles:
+    - values.yaml
+    variables:
+      - name: REPLICA_COUNT
+        description: "Override the number of pod replicas"
+        path: replicaCount
+  dataInjections:
+  - source: zim-data
+    target:
+      namespace: my-namespace
+      selector: app=my-app
+      container: data-loader
+      path: /data
+    compress: true
+  files:
+  - source: source-file.txt
+    target: target-file.txt
+    shasum: shasum
+    executable: false
+    symlinks:
+    - /path/to/symlink
+    extractPath: /path/to/extract
+  images:
+  - podinfo@v1
+  repos:
+  - https://github.com/defenseunicorns/zarf
+  extensions:
```

Comment:
I think we need to pull extensions out of zarf before 1.0. I know we've discussed before, but I cant seem to find an issue on it now.

Reply:
I don't think we have an issue specifically for bigbang, but you may be referring to this issue https://github.com/defenseunicorns/zarf/issues/1136

Reply:
Yeah that never happened obviously so this is one of those big things we should move to the SDK model prior to 1.0.

Reply:
Yeah I was tracking it against #1136 and agree it should get smushed out of the base schema